### ISSUE TITLE
Clarify and Expand the Team Workflow Diagram 

### ISSUE THREAD

**Issue by nelsonic** [tcm-labs/workflow#9]
The "_Old_" (Word Document) of 
included a diagram entitled "_Simple Issue Tracking Workflow_":

![simple-issue-tracking-workflow](https://user-images.githubusercontent.com/194400/31656162-c04e50fa-b322-11e7-8950-18c4913188d8.png)

## Todo

+ [ ] Revise the diagram in light of the change in our workflow.
+ [ ] Clarify exactly what stage in the workflow a task/activity relates to
+ [ ] Show who is responsible for the stage

**Comment by nelsonic** [comment]
First Draft:
![tcm-workflow](https://user-images.githubusercontent.com/194400/31734689-72585bd2-b437-11e7-9ce4-9431f516efe9.png)
edit this Google Doc: https://goo.gl/2UNVbw (_using [draw.io](https://www.draw.io)_)

**Comment by nelsonic** [comment]
With Blockers:
![tcm-workflow-with-blockers](https://user-images.githubusercontent.com/194400/31735912-26fa7892-b43b-11e7-9da1-75547a348f89.png)

**Comment by maniananth** [comment]
Hi Nelson,

First draft looks good.

please find my comments below

1.Ideation
  - what if the Ideation is a Technical Story or Task and not a Product Feature? How does that flow?
  - Should there be a mention that only the respective stage owner or a proxy should move the story to the next stage in the process
 - An example issue would be great for this stage 
2.Estimate : 
  - Do stories / tasks need to be broken down as Front end and Backend tasks? 
  - Waffle / Github do not have any placeholder for entering estimates? Where should these be held?
  - Should the sub-tasks ( if story is too big and needs broken down ) be connected to the main story and be estimated individually?
  - An example issue would be great for this stage 
  - Do we need the testers to estimate as well?
3 .Assign
 - while the dev is assigning an issue to themselves , should the tester assign a testing task to themselves  and start writing the test cases / auto - tests?
  - An example issue would be great for this stage 
4.Build
 - Is the Blocked indicator a label in github? that goes against the issue
  - An example issue would be great for this stage 
5. Finalise
  - In case of subtasks , should all connected PR's be completed before the issue is sent to review? or can individual subtasks be sent to review?
6.Ready for Test
 - In case of Technical Stories , what is the approach to testing and who does it?
7.Testing
 - Should  there be a Blocked status for Testing too?
8.Should reported bugs need triage first and then if proven needs prioritization from the PO? How do we capture this in the process?
9.Should the feature be tagged to a Release Number / version 

Thanks and Regards,
Mani Ananth

**Comment by michaellorman** [comment]
Hi @nelsonic, couple of questions from me:
1. We don`t have Waffle statuses on the diagram. Can we add them?
2. Also in Waffle there is no possibility to create sub-tasks under user story, so sub-tasks will be created by devs as separate tickets. Tickets created by devs probably won`t have such description as we have in user story written by POs. So this sub-tasks won`t be tested by QAs. So can we create additional PRs under original user story and don`t create separate tickets as sub-tasks?
3. Also ticket can be blocked on the stage of testing, not only on development stage.

**Comment by hwndept** [comment]
the flow looks good to me, great work @nelsonic! I like the kind of visualisation you use  👍

@michaellorman, Nelson highlighted that dev team may return the story back to BA/FA if it's too big, so, basically, it shouldn't be a case. Also, tech/lead and/or dev team should create technical criteria for every story, if team feels confident to deliver the entire story by creating several PRs - alls PRs can be explicitly linked to every tech criteria, e.g.:

###### Tech Criteria
- [ ] Build new screen(#9);
- [ ] Provide the API(#9);
- ...
+1 to point to a case when **testing is blocked**

Thanks!

**Comment by hwndept** [comment]
Guys and another point from my side, I would prefer to avoid estimates, the better approach for Kanban is to have stories approximately similar complexity and track lead time and circle time - this will allow us to forecast the delivery.

**Comment by nelsonic** [comment]
@hwndept the question of _estimation_ has it's _own_ separate issue: https://github.com/tcm-labs/workflow/issues/8 
Estimation is a _balance_ between "_overhead_" and "_accountability_",
I _always_ tend towards more ***accountability*** 

Right now we have stories that have "**Lead time**" of 2+ months. (_because the story was created ages ago_) but no (_Dev_) work has been accrued.

We also have stories that have "**Cycle Time**" of a couple of _weeks_ because "_short bursts_" of work have been done on them but they are still not "_done_".

For example: The _current_ milestone https://github.com/tcm-labs/tcm-payg-app/milestone/3
is _meant_ to be complete **Today** and yet only **9%** of stories are "_done_":
![image](https://user-images.githubusercontent.com/194400/32297157-2a1f63f0-bf46-11e7-837a-7794017eed86.png)

If people are _required_ to estimate how long each story is _likely_ to take the Project Manager, Team Lead and Product Owner would have a _much_ clearer picture of when to **`expect`** the story to be `done` and can "_manage expectations_" of "The Business" ("_stakeholders_") a lot better.
